ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Basics.hs

```diff
@@ -88,11 +88,16 @@ newtype instance LedgerState (HardForkBlock xs) mk = HardForkLedgerState {
       hardForkLedgerStatePerEra :: HardForkState (Flip LedgerState mk) xs
     }
 
-deriving stock   instance CanHardFork xs                => Eq       (LedgerState (HardForkBlock xs) mk)
-deriving newtype instance (CanHardFork xs, Typeable mk) => NoThunks (LedgerState (HardForkBlock xs) mk)
+deriving stock   instance CanHardFork xs => Eq       (LedgerState (HardForkBlock xs) EmptyMK)
+deriving newtype instance CanHardFork xs => NoThunks (LedgerState (HardForkBlock xs) EmptyMK)
 
-instance (SingI mk, CanHardFork xs) => Show (LedgerState (HardForkBlock xs) mk) where
-  showsPrec p = showParen (p >= 11) . showsLedgerState sing
+deriving stock   instance CanHardFork xs => Eq       (LedgerState (HardForkBlock xs) ValuesMK)
+deriving newtype instance CanHardFork xs => NoThunks (LedgerState (HardForkBlock xs) ValuesMK)
+
+deriving newtype instance CanHardFork xs => NoThunks (LedgerState (HardForkBlock xs) SeqDiffMK)
+
+instance (SingI mk, CanHardFork xs) => Show (LedgerState (HardForkBlock xs) (ApplyMapKind' mk)) where
```

Comment:
Why didn't we need to replace `SingI` by `IsApplyMapKind` here? :thinking:

Reply:
I could have, but then it would have required `UndecidableInstances` which is one of the few LANGUAGE pragmas I'm hesitant to introduce without good cause. So what I'm done here is _almost_ equivalent but doesn't frighten GHC.